### DISCUSSION TITLE
test HA Supervised@Docker su sistema con HA@Docker

### DISCUSSION THREAD

**Discussion by labattazza** [indomus/forum#2174] [HUB: Home Assistant]
Buondì a tutti.

Ho letto con interesse l'articolo a proposito dell'installazione di **HA Supervised@Docker**.

Siccome in questo periodo "la frenesia della vita moderna" fa si che non abbia molto tempo a disposizione per fare debug approfondito riguardo all'installazione di alcuni componenti che mi creano problemi, volevo provare questa modalità, principalmente per testare la comodità dello store (e in parte del backup, anche se ora me la cavo con alcuni script che mi copiano le configurazioni su una condivisione di rete).

Attualmente **ho una installazione HA@docker tramite docker compose** (idem per MQTT, InfluxDb etc).

Prima di partire volevo capire come evitare conflitti: **è sufficiente stoppare i container** attualmente in uso, o è necessario eliminarli del tutto?
Una ultima cosa che non mi è chiara: si può **definire una directory per i file di configurazione**, ad esempio nella home, o vengono salvati tutti nella directory del container?

Grazie mille!

**Comment by snakuzzo** [comment]
Innanzi tutto, devi assicurarti di avere Debian (non altre distro derivate...deve essere proprio Debian).
Quanto alla dir di configurazione che verrà usata, non ricordo quale sia...ma quasi sicuramente non è quella che stai usando attualmente tu. Se non sbaglio viene creata sotto `/share`, ma va visto nello script di installazione. Ad ogni modo o si cambia nello script (ma meglio evitare) o fai in modo di creare un symlink, così la piazzi dove ti pare (questo per dire che comunque quando viene creato il container di core viene fatto un bind con una dir sull'host altrimenti, ogni volta che ricreeresti il container per un aggiornamento, perderesti tutto)

**Reply by snakuzzo** [reply]
Ho dato un'occhiata proprio ora alla repo github ufficiale e nella versione 1.4.1 rilasciata due giorni fa c'è anche l'indicazione per rimappare la dir da usare per la creazione delle dir di configurazione. Leggi [QUI](https://github.com/home-assistant/supervised-installer#configuration)

**Reply by labattazza** [reply]
Grazie mille, andrò a leggere sul repo allora.
Visto che sto meditando di cambiare server e passare da un i5 4°generazione (consumo sui 25watt e qualche incertezza a volte nel vedere filmati in streaming se un processo in background occupa più risorse) a uno dei vari mini-pc.
Così reinstallando da zero metto Debian, e via 
PR: leggevo che gli sviluppatori sconsigliano caldamente di usare la macchina per altro che non sia HA supervised. Io avevo intenzione di metter anche uno stack LAMP per usare Nextcloud, e mi pare che anche nell vostre prove aveste installato altro software, dico bene?

**Reply by snakuzzo** [reply]
Puoi fare quello che vuoi, l'importante è che non si vadano ad installare pacchetti che vadano in conflitto con i prerequisiti di HA Supervised.
Puoi leggerlo [QUI](https://github.com/home-assistant/supervised-installer#install-home-assistant-supervised).
Loro (i dev) giustamente non possono stare lì a darti supporto se il tuo ambiente è pieno di altra "robaccia", quindi in caso di problemi ed eventuali issue su github, non ti daranno alcun supporto.
Resta il fatto che... ok le guide, ok provare ... ma è un tipo di installazione sconsigliata a chi non è pratico...cito...

> This method is considered advanced and should only be used if one is an expert in managing a Linux operating system, Docker and networking.

**Reply by labattazza** [reply]
Si, certo, loro logicamente non possono garantire altro se non la configurazione sulla quale hanno ottimizzato il software, e stare dietro alle miriadi di varianti che ogni utente può presentare sarebbe impossibile.
Non sono un sysadmin di professione, un po' di esperienza me la sono fatta e spesso riesco a risolvermi i problemi che si presentano. 
Approfitto della necessità di fare una installazione da zero per usare la cara vecchia "mamma Debian" lasciando ubuntu sul NUC tuttofare, e sperimento HA Supervised. 
Poi sono sempre in tempo a ritornare a HA@Docker. Tra l'altro, in questo modo sperimento anche la migrazione di configurazione HA@Docker==>HaSupervised@Docker, mi seccherebbe dover rifare il pairing di tutti i dispositivi.
Chissà se riuscirò a migrare anche lo storico di InfluxDB?

Grazie mille per la tua opinione

**Reply by labattazza** [reply]
Ciao

aggiorno giusto per portarvi la mia esperienza con questa modalità.

Nel server avevo già installato uno stack LAMP per far girare nextcloud, che funziona.
Nei ritagli di tempo ho installato HA Supervised@Docker, e ha funzionato.

Ho avuto qualche piccolo intoppo coi vari addon, la maggior parte risolti controllando i log (ad esempio per il terminale ssh dovevo cambiare la porta, visto che la 22 la usavo già per connettermi da remoto con ssh "classico").

Ogni tanto, però, qualche "stranezza" appare. Ad esempio prima "glances" e poi Grafana non ne volevano sapere di avviarsi. Premevo su "avvia", e dopo un lampo restavano "spenti".
La cosa particolare, che ho verificato è capitata anche ad altri, era che il container risultava avviato correttamente, sia leggendo i log che controllando i container avviati.

Per Grafana il problema si era presentato a pranzo, sono dovuto uscire e rientrare poco fa: controllo e.. Grafana è attivo! E' come se ci fosse un ritardo all'avvio.

Ora proseguo, prima di "spegnere" il server su cui girava voglio verificare se riesco a trasferire la configurazione da Home Assistant @ Docker a Home Assistant SUpervised @Docker: ho trovato solo indicazioni su come passare da Core/Venv a Docker. 
Più che altro mi piacerebbe trasferire i dati di due mesi di misurazioni confluiti in InfluxDB, e ho qualche problema coi comandi da terminale per le query.

Ma più tardi e domani studio un po' la situazione. 
Se qualcuno avesse sotto mano delle buone dritte.. accetto volentieri! Altrimenti se butta male chiedo aiuto, o riconfiguro da zero approffittandone per sistemare un paio di sviste.

**Reply by snakuzzo** [reply]
Semplicemente...passi la config dir di core su docker sull'altra installazione.
Fine
